components/tel/Core/Inc/radio.h

```diff
@@ -43,6 +44,17 @@
 /* TIMING CONSTANTS */
 #define CAN_TRANSMIT_TIMEOUT                1000                // 1 second timeout
 
+/*XBee API Packet CONSTANTS*/
+#define API_BUFFER_SIZE 300 //number of bytes in api_buffeer array
```

Comment:
Currently, you will send extra null characters in the api_buffer because your api messages will be (10 CAN Messages) * (24 bytes per CAN message) + 18 bytes overhead = 258 bytes total. 300 - 258 = 42 bytes of null characters. This problem with this is that sunlink's process_message algorithm will take these null characters and assume they are a piece of the next api_message (since it could not cut it up into a valid message). Do your sunlink changes resolve this?

Reply:
300 is just the API buffer size. When I create the API packet array, it is based off the api_buffer position (which marks the end of can messages in the buffer) + the api overhead size.  This means, the api packet should be exactly the right size and no null characters will be transmitted. On sunlink side, if null characers were transmitted, we would get an error because the parser would try to parse groups of 24 null characters as a can message. This can be changed though using the length field (number of can messages) we transmit.

Reply:
Right that makes sense. I see your code that sets the api_buffer to the value of the api_buffer_position. Thanks for clarifying!